Blocstagram/Blocstagram/ImagesTableViewController.m

```diff
@@ -63,15 +56,15 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
         [cell.contentView addSubview:imageView];
     }
     
-    UIImage *image = self.images[indexPath.row];
-    imageView.image = image;
+    Media *media = [self items][indexPath.row];
+    imageView.image = media.image;
     
     return cell;
 }
 
 - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
-    UIImage *image = self.images[indexPath.row];
-    return (CGRectGetWidth(self.view.frame) / image.size.width) * image.size.height;
+    Media *media = [self items][indexPath.row];
+    return (CGRectGetWidth(self.view.frame) / media.image.size.width) * media.image.size.height;
```

Comment:
Just one quick note, wherever you wind up dividing by something that isn't a constant (in this case, `media.image.size.width`), I would make sure that that value is not zero before doing the math...